<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1369</width>
    <height>1001</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1369</width>
    <height>1001</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1369</width>
    <height>1001</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>s21_3DViewer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="modelView" name="model_view" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>941</width>
      <height>941</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_OpenFile">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>40</y>
      <width>101</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string>Open file</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_modelInfo">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>40</y>
      <width>251</width>
      <height>71</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_fileFormat">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>890</y>
      <width>191</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>jpg</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>bmp</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="snapshotButton">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>810</y>
      <width>191</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string>Snapshot</string>
    </property>
   </widget>
   <widget class="QLabel" name="x_shift">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>550</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>x shift</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="y_shift">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>550</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>y shift</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="z_shift">
    <property name="geometry">
     <rect>
      <x>1150</x>
      <y>550</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>z shift</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="zShiftInput">
    <property name="geometry">
     <rect>
      <x>1150</x>
      <y>580</y>
      <width>71</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-2.000000000000000</double>
    </property>
    <property name="maximum">
     <double>2.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="xShiftInput">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>580</y>
      <width>71</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-2.000000000000000</double>
    </property>
    <property name="maximum">
     <double>2.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="yShiftInput">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>580</y>
      <width>71</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-2.000000000000000</double>
    </property>
    <property name="maximum">
     <double>2.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QPushButton" name="shiftButton">
    <property name="geometry">
     <rect>
      <x>1230</x>
      <y>550</y>
      <width>121</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>SHIFT</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_projectionType">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>330</y>
      <width>111</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Parallel</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Central</string>
     </property>
    </item>
   </widget>
   <widget class="QDoubleSpinBox" name="xTurnInput">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>500</y>
      <width>71</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-2.000000000000000</double>
    </property>
    <property name="maximum">
     <double>2.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="yTurnInput">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>500</y>
      <width>71</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-2.000000000000000</double>
    </property>
    <property name="maximum">
     <double>2.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="zTurnInput">
    <property name="geometry">
     <rect>
      <x>1150</x>
      <y>500</y>
      <width>71</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-2.000000000000000</double>
    </property>
    <property name="maximum">
     <double>2.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton">
    <property name="geometry">
     <rect>
      <x>1240</x>
      <y>470</y>
      <width>106</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Radian</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="radioButton_2">
    <property name="geometry">
     <rect>
      <x>1240</x>
      <y>500</y>
      <width>106</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Grade</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup</string>
    </attribute>
   </widget>
   <widget class="QDoubleSpinBox" name="scaleInput">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>660</y>
      <width>105</width>
      <height>26</height>
     </rect>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QPushButton" name="yTurnButton">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>470</y>
      <width>71</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate y</string>
    </property>
   </widget>
   <widget class="QPushButton" name="zTurnButton">
    <property name="geometry">
     <rect>
      <x>1150</x>
      <y>470</y>
      <width>71</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate z</string>
    </property>
   </widget>
   <widget class="QPushButton" name="scaleButton">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>630</y>
      <width>106</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Scale</string>
    </property>
   </widget>
   <widget class="QPushButton" name="xTurnButton">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>470</y>
      <width>71</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate x</string>
    </property>
   </widget>
   <widget class="QPushButton" name="resizeButton">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>700</y>
      <width>371</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Resize</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_backColor">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>290</y>
      <width>231</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Background color</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_vertexColor">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>190</y>
      <width>111</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertex color</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_edgeColor">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>240</y>
      <width>111</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Edge color</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="pointSizeInput">
    <property name="geometry">
     <rect>
      <x>1310</x>
      <y>190</y>
      <width>44</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>15</number>
    </property>
    <property name="value">
     <number>5</number>
    </property>
   </widget>
   <widget class="QLabel" name="point_size">
    <property name="geometry">
     <rect>
      <x>1220</x>
      <y>190</y>
      <width>71</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Point size</string>
    </property>
   </widget>
   <widget class="QComboBox" name="vertexType">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>190</y>
      <width>101</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>No vertex</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Square vertex</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Circle vertex</string>
     </property>
    </item>
   </widget>
   <widget class="QSpinBox" name="edgeSize">
    <property name="geometry">
     <rect>
      <x>1310</x>
      <y>240</y>
      <width>44</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>15</number>
    </property>
   </widget>
   <widget class="QLabel" name="line_width">
    <property name="geometry">
     <rect>
      <x>1220</x>
      <y>240</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Line width</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_edgesType">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>240</y>
      <width>101</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Solid line</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Dotted line</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="pushButton_saveOptions">
    <property name="geometry">
     <rect>
      <x>1210</x>
      <y>290</y>
      <width>141</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Save options</string>
    </property>
   </widget>
   <widget class="QLabel" name="projection">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>330</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Projection</string>
    </property>
   </widget>
   <widget class="QLabel" name="focusDistatnce">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>380</y>
      <width>101</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Focus distance</string>
    </property>
   </widget>
   <widget class="QSlider" name="scaleInputSlider">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>650</y>
      <width>251</width>
      <height>20</height>
     </rect>
    </property>
    <property name="maximum">
     <number>20</number>
    </property>
    <property name="singleStep">
     <number>1</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>10</number>
    </property>
    <property name="sliderPosition">
     <number>10</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="gifMakeButton">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>810</y>
      <width>171</width>
      <height>101</height>
     </rect>
    </property>
    <property name="text">
     <string>GIF</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>630</y>
      <width>16</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>1210</x>
      <y>630</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>1</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>1330</x>
      <y>630</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>2</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox_focus_dist">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>370</y>
      <width>111</width>
      <height>30</height>
     </rect>
    </property>
    <property name="minimum">
     <double>1.000000000000000</double>
    </property>
    <property name="maximum">
     <double>100.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QPushButton" name="restoreOptionsButton">
    <property name="geometry">
     <rect>
      <x>1210</x>
      <y>350</y>
      <width>141</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Restore options</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>140</y>
      <width>381</width>
      <height>16</height>
     </rect>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>430</y>
      <width>381</width>
      <height>16</height>
     </rect>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>760</y>
      <width>381</width>
      <height>16</height>
     </rect>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1369</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>modelView</class>
   <extends>QWidget</extends>
   <header>modelview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
